### ISSUE TITLE
osx + rbenv default setup + lolcommits: tests fail

### ISSUE THREAD

**Issue by kenmazaika** [mroth/lolcommits#100]
This issue took me a long time to debug.  The issue is bizarre, and I'm not sure what the best course of action is to resolve this, but I figured I'd make a note here so someone else doesn't spend the time shaving yaks that I just did.

The problem was:
- I installed lolcommits, worked great locally
- Ran tests and they failed.  Womp, womp.

The failure message looked like this:

```
create mode 100644 voluptas
/System/Library/Frameworks/Ruby.framework/Versions/1.8/usr/lib/ruby/1.8/rubygems/custom_require.rb:31:in `gem_original_require': no such file to load -- mini_magick (LoadError)
    from /System/Library/Frameworks/Ruby.framework/Versions/1.8/usr/lib/ruby/1.8/rubygems/custom_require.rb:31:in `require'
    from /Users/kenmazaika/source/lolcommits/bin/lolcommits:11
```

I spent some time debugging why it couldn't find mini_magick, and it turned out it was using the wrong version of ruby.  Also note, I installed rbenv per the docs in rbenv and used ruby-build.

In rbenv I have:

```
1.9.3-p327    < global
2.0.0-p0      < (set by /Users/kenmazaika/source/.rbenv-version)
1.8.7         < system ruby
```

Digging into it, it was picking up the system ruby when running them in the test.

I added a sleep right before the tests failed, and when manually executing the command in the terminal, the expected behavior happened, the problems only existed in the test.  Eventually I stumbled upon the problem:
#### Long Story Short; git post commit hooks garble the `ENV['PATH']` only if the git command is shelled out to from ruby.

```
Kens-MacBook-Pro:~ kenmazaika$ cd /tmp/
Kens-MacBook-Pro:tmp kenmazaika$ mkdir omg
Kens-MacBook-Pro:tmp kenmazaika$ cd omg/
Kens-MacBook-Pro:omg kenmazaika$ git init .
Initialized empty Git repository in /private/tmp/omg/.git/

Kens-MacBook-Pro:omg kenmazaika$ curl https://gist.github.com/kenmazaika/5479182/raw/823eb3d6f2affbea7da41d2b539780e828d9c417/post-commit >> .git/hooks/post-commit

Kens-MacBook-Pro:omg kenmazaika$ ruby -v
ruby 1.9.3p327 (2012-11-10 revision 37606) [x86_64-darwin12.1.1]
Kens-MacBook-Pro:omg kenmazaika$ chmod a+x .git/hooks/post-commit
Kens-MacBook-Pro:omg kenmazaika$ git commit --allow-empty -m 'can haz commit'
/Users/kenmazaika/.rbenv/versions/1.9.3-p327/bin
/Users/kenmazaika/.rbenv/libexec
/Users/kenmazaika/.rbenv/plugins/ruby-build/bin
/usr/libexec/git-core
/usr/bin
/Users/kenmazaika/.rbenv/shims
/usr/local/bin
/usr/local/sbin
/usr/local/heroku/bin
/Users/kenmazaika/.rbenv/shims
/Users/kenmazaika/.rbenv/bin
/usr/bin
/bin
/usr/sbin
/sbin
/usr/local/bin
/usr/local/git/bin
[master 9854ee3] can haz commit


Kens-MacBook-Pro:omg kenmazaika$ irb
irb(main):001:0> `git commit --allow-empty -m 'can haz commit'`
/usr/libexec/git-core
/usr/bin
/Users/kenmazaika/.rbenv/versions/1.9.3-p327/bin
/Users/kenmazaika/.rbenv/libexec
/Users/kenmazaika/.rbenv/plugins/ruby-build/bin
/Users/kenmazaika/.rbenv/shims
/usr/local/bin
/usr/local/sbin
/usr/local/heroku/bin
/Users/kenmazaika/.rbenv/shims
/Users/kenmazaika/.rbenv/bin
/usr/bin
/bin
/usr/sbin
/sbin
/usr/local/bin
/usr/local/git/bin
irb(main):003:0> puts ENV['PATH'].split(":").join("\n")
/Users/kenmazaika/.rbenv/versions/1.9.3-p327/bin
/Users/kenmazaika/.rbenv/libexec
/Users/kenmazaika/.rbenv/plugins/ruby-build/bin
/Users/kenmazaika/.rbenv/shims
/usr/local/bin
/usr/local/sbin
/usr/local/heroku/bin
/Users/kenmazaika/.rbenv/shims
/Users/kenmazaika/.rbenv/bin
/usr/bin
/bin
/usr/sbin
/sbin
/usr/local/bin
/usr/local/git/bin
```

Note how if call the git command directly, `/usr/libexec/git-core`, `/usr/bin` are toward the bottom of the `PATH` but when called through ruby, they mysteriously come to the top, even though in the containing ruby `ENV['PATH']` also has these lines lower in the path.
#### TLDR; move /usr/bin/ruby into /usr/bin/ruby-old if you are using rbenv and having a hard time getting tests to run.

![public service announcement](http://mazaika.io/ruby-bad-time)

Might be nice to add a unit test that spits out some sort of a warning message if the following is true:

```
ENV['PATH'].match(/rbenv/) && File.exists?('/usr/bin/ruby')
```
